### PR TITLE
Update references to new API doc site.

### PR DESCRIPTION
**Pull request by Jeff-Meadows**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @Jeff-Meadows has signed the CLA. Thanks for the pull request!

File: boxsdk/object/base_object.py [hunk 1]
```diff
@@ -117,7 +117,7 @@ def update_info(self, data, params=None, headers=None, **kwargs):
             Update the object attributes in data.keys(). The semantics of the
             values depends on the the type and attributes of the object being
             updated. For details on particular semantics, refer to the Box
-            developer API documentation <developers.box.com/docs>.
+            developer API documentation <https://box-content.readme.io/docs/>.
```
Comment (jmoldow) [inline boxsdk/object/base_object.py]: Is this what we want? That leads to a getting started page. Should we link to https://box-content.readme.io/reference, or even just to https://box-content.readme.io (which redirects to https://box-content.readme.io/reference) instead?

**Comment by jmoldow** [comment]
Looks good, just one question.

**Comment by jmoldow** [comment]
@Jeff-Meadows do you want to rebase this and include it in the release?

**Pull request merged**